tests/app/services/model_records/test_model_records_sql.py

```diff
@@ -0,0 +1,210 @@
+    DuplicateModelException,
+    ModelRecordServiceBase,
+    ModelRecordServiceSQL,
+    UnknownModelException,
+)
+from invokeai.app.services.shared.sqlite import SqliteDatabase
+from invokeai.backend.model_manager.config import (
+    BaseModelType,
+    DiffusersConfig,
+    ModelType,
+    TextualInversionConfig,
+    VaeDiffusersConfig,
+)
+from invokeai.backend.util.logging import InvokeAILogger
+
+
+@pytest.fixture
+def store(datadir) -> ModelRecordServiceBase:
+    config = InvokeAIAppConfig(root=datadir)
+    logger = InvokeAILogger.get_logger(config=config)
+    db = SqliteDatabase(config, logger)
+    return ModelRecordServiceSQL(db)
+
+
+def example_config() -> TextualInversionConfig:
+    return TextualInversionConfig(
+        path="/tmp/pokemon.bin",
+        name="old name",
+        base=BaseModelType("sd-1"),
+        type=ModelType("embedding"),
+        format="embedding_file",
+        original_hash="ABC123",
+    )
+
+
+def test_add(store: ModelRecordServiceBase):
+    raw = dict(
+        path="/tmp/foo.ckpt",
+        name="model1",
+        base=BaseModelType("sd-1"),
+        type="main",
+        config="/tmp/foo.yaml",
+        variant="normal",
+        format="checkpoint",
+        original_hash="111222333444",
+    )
+    store.add_model("key1", raw)
+    config1 = store.get_model("key1")
+    assert config1 is not None
+    assert config1.base == BaseModelType("sd-1")
+    assert config1.name == "model1"
+    assert config1.original_hash == "111222333444"
+    assert config1.current_hash is None
+
+
+def test_dup(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", example_config())
+    with pytest.raises(DuplicateModelException):
+        store.add_model("key1", config)
+    with pytest.raises(DuplicateModelException):
+        store.add_model("key2", config)
+
+
+def test_update(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    config = store.get_model("key1")
+    assert config.name == "old name"
+
+    config.name = "new name"
+    store.update_model("key1", config)
+    new_config = store.get_model("key1")
+    assert new_config.name == "new name"
+
+
+def test_unknown_key(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    with pytest.raises(UnknownModelException):
+        store.update_model("unknown_key", config)
+
+
+def test_delete(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    config = store.get_model("key1")
+    store.del_model("key1")
+    with pytest.raises(UnknownModelException):
+        config = store.get_model("key1")
+
+
+def test_exists(store: ModelRecordServiceBase):
+    config = example_config()
+    store.add_model("key1", config)
+    assert store.exists("key1")
+    assert not store.exists("key2")
+
+
+def test_filter(store: ModelRecordServiceBase):
+    config1 = DiffusersConfig(
```

Comment:
I think this should be `MainDiffusersConfig`?

Reply:
I've changed all the classes to be the terminal ones.